gr-blocks/lib/add_const_ff_impl.cc

```diff
@@ -52,22 +53,16 @@ int add_const_ff_impl::work(int noutput_items,
 
     int size = noutput_items;
 
-    while (size >= 8) {
-        *optr++ = *iptr++ + d_k;
-        *optr++ = *iptr++ + d_k;
-        *optr++ = *iptr++ + d_k;
-        *optr++ = *iptr++ + d_k;
-        *optr++ = *iptr++ + d_k;
-        *optr++ = *iptr++ + d_k;
-        *optr++ = *iptr++ + d_k;
-        *optr++ = *iptr++ + d_k;
-        size -= 8;
+    // build column vec. d_k * ones(size, 1) for volk addition
```

Comment:
This whole section of code should be moved to the `set` method, which will have to be moved from the header `gr-blocks/lib/add_const_ff_impl.h` to this file. And then it will have to be called at instantiation as well. At instantiation declaration, set the copy_vec to NULL. Then in `set` check whether the copy_vec is NULL & if not then `volk_free` it. Same check in the destructor. Not trivial, but it'll clean up this code and make it faster and safer!